drivers/leds/Kconfig

```diff
@@ -123,6 +123,22 @@ config WS2812_NON_SPI_DRIVER
 		implementations.  Selecting this option builds the new
 		model driver.
 
+config WS2812_RMT_OUTPUT_PIN
+	int "The pin where the ws2812 are connected."
+	depends on WS2812_NON_SPI_DRIVER
+	default 4
+	range 1 35
+	---help---
+		The GPIO pin where the ws2812 are connected.
```

Comment:
It is not appropriate for mapping GPIO pins via Kconfig, which is a board characteristic that should be immutable.
As an example, see how the LCD display of the [ESP-WROVER-KIT board](https://nuttx.apache.org/docs/latest/platforms/xtensa/esp32/boards/esp32-wrover-kit/index.html) is mapped on its interface: https://github.com/apache/incubator-nuttx/blob/c5785ee9d5cf6d6ee7626316361a4735a0ee5195/boards/xtensa/esp32/esp32-wrover-kit/include/board.h#L42-L47